File: grafos/trabalhos/2010-2/trab2/calj08-rbc08/src/Vertice.java
public class Vertice implements Cloneable
{
	private boolean visitado;
	private int nome;
	private int grau;
	
	
	public Vertice() {}
	/*---------------------------------*/
	
	
	public Vertice( int nome )
	{
		this.setNome( nome );
	}
	/*------------------------------------------*/
	
	
	public Vertice( int nome, boolean visitado )
	{
		this.setNome( nome );
		this.setVisitado( visitado );
	}
	/*--------------------------------*/
	
	
	public Vertice( int nome, int grau, boolean visitado )
	{
		this.setNome( nome );
		this.setGrau( grau );
		this.setVisitado( visitado );
	}
	
	
	/**
	 * Sobrescreve o método clone(), de Object
	 */
	public Object clone()
	{
		try
		{
			return super.clone();
		}
		catch( CloneNotSupportedException e )
		{
			System.err.println( "Clonagem não suportada: " + e.getMessage() );
			return null;
		}
	}
	/*------------------------------------*/
	
	
	public boolean isVisitado()
	{
		return this.visitado;
	}
	/*------------------------------------------*/
	
	
	
	public void setVisitado( boolean visitado )
	{
		this.visitado = visitado;
	}
	/*------------------------------------------*/
	
	
	public int getNome()
	{
		return this.nome;
	}
	/*--------------------------------------*/
	
	
	public void setNome( int nome )
	{
		this.nome = nome;
	}
	/*-------------------------------------------*/
	
	
	public int getGrau()
	{
		return this.grau;
	}
	/*---------------------------------------------*/
	
	
	public void setGrau( int grau )
	{
		this.grau = grau;
	}
}

File: grafos/trabalhos/2010-2/trab1/calj08-rbc08/src/lib/vertice.cpp
#include "vertice.h"

Vertice::Vertice() {}
/*-----------------------------------------*/


Vertice::Vertice( unsigned int nome )
{
	this->setNome( nome );
}
/*--------------------------------------------------*/


Vertice::Vertice( unsigned int nome, bool visitado )
{
	this->setNome( nome );
	this->setVisitado( visitado );
}
/*--------------------------------------------------*/

/* fp é opcional */
void Vertice::exibir( FILE *fp )
{
	fprintf( fp, "%d", this->getNome() );
}
/*--------------------------------------*/


void Vertice::setVisitado( bool visitado )
{
	this->visitado = visitado;
}
/*--------------------------------------*/



void Vertice::setNome( unsigned int nome )
{
	this->nome = nome;
}
/*-------------------------------*/




bool Vertice::isVisitado()
{
	return this->visitado;
}
/*---------------------------------------------*/



unsigned int Vertice::getNome()
{
	return this->nome;
}
/*---------------------------------------*/


